pkg/storage/edv/restprovider.go

```
@@ -0,0 +1,404 @@
+
+	_, err = r.restClient.createDocument(r.vaultID, &encryptedDocument)
+	if err != nil {
+		return fmt.Errorf(failCreateDocumentInEDVServer, err)
+	}
+
+	return nil
+}
+
+func (r *restStore) Get(k string) ([]byte, error) {
+	indexValueMACBase64Encoded, err := r.computeIndexValueMACBase64Encoded(k)
+	if err != nil {
+		return nil, fmt.Errorf(failComputeBase64EncodedIndexValueMAC, err)
+	}
+
+	matchingDocumentIDs, err := r.restClient.queryVault(r.vaultID, &Query{
+		Name:  r.indexKeyMACBase64Encoded,
+		Value: indexValueMACBase64Encoded,
+	})
+	if err != nil {
+		return nil, fmt.Errorf(failQueryVaultInEDVServer, err)
+	}
+
+	if len(matchingDocumentIDs) == 0 {
+		return nil, fmt.Errorf("no documents matching the query were found: %w", storage.ErrDataNotFound)
+	} else if len(matchingDocumentIDs) > 1 {
+		// This should only be possible if the EDV server is not able to maintain the uniqueness property of the
+		// indexedAttribute created in the createIndexedAttribute method.
+		// TODO: Check each of the documents to see if they all have the same content (other than the document ID).
+		// If so, we should delete the extras and just return one of them arbitrarily.
+		return nil, errors.New("multiple documents matching the query were found." +
+			"This probably indicates an issue with the EDV server's database")
+	}
+
+	encryptedDocumentBytes, err := r.restClient.readDocument(r.vaultID, matchingDocumentIDs[0])
+	if err != nil {
+		return nil, fmt.Errorf(failRetrieveDocumentFromEDVServer, err)
+	}
+
+	return encryptedDocumentBytes, nil
+}
+
+// Not support by restStore. The EDV data vault HTTPS API doesn't support this type of operation,
+// nor is there a feasible way to emulate it. The iterator returned always is always in an error state.
+func (r *restStore) Iterator(string, string) storage.StoreIterator {
+	return &restStoreIterator{}
+}
+
+func (r *restStore) Delete(string) error {
+	panic("implement me")
```

Should return an error here.

---

Fixed. I also added a TODO above the method since Delete is actually in the EDV REST API - we just don't have a use case for it and it's not supported in the implementation I'm testing it against... yet.

https://github.com/hyperledger/aries-framework-go/issues/2286